src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeLift.java

```diff
@@ -418,73 +122,332 @@ public static CoreOps.FuncOp liftToBytecodeDialect(byte[] classdata, String meth
-                                exceptionTableStart(c.blockMap.get(handler).successor()));
-                    } else if (labelTarget.label() == tryCatchBlock.tryEnd()) {
-                        b.op(BytecodeInstructionOps.exceptionTableEnd());
+                        }
+                        case NewReferenceArrayInstruction inst -> {
+                            stack.push(b.op(CoreOps.newArray(
+                                    TypeDesc.type(TypeDesc.ofNominalDescriptor(inst.componentType().asSymbol()), 1),
+                                    stack.pop())));
+                        }
+                        case NewMultiArrayInstruction inst -> {
+                            stack.push(b.op(CoreOps._new(
+                                    MethodTypeDesc.methodType(
+                                            TypeDesc.ofNominalDescriptor(inst.arrayType().asSymbol()),
+                                            Collections.nCopies(inst.dimensions(), TypeDesc.INT)),
+                                    IntStream.range(0, inst.dimensions()).mapToObj(_ -> stack.pop()).toList().reversed())));
+                        }
+                        case TypeCheckInstruction inst when inst.opcode() == Opcode.CHECKCAST -> {
+                            stack.push(b.op(CoreOps.cast(TypeDesc.ofNominalDescriptor(inst.type().asSymbol()), stack.pop())));
+                        }
+                        case StackInstruction inst -> {
+                            switch (inst.opcode()) {
+                                case POP, POP2 -> stack.pop(); // @@@ check the type width
+                                case DUP, DUP2 -> stack.push(stack.peek());
+                                //@@@ implement all other stack ops
+                                default ->
+                                    throw new UnsupportedOperationException("Unsupported stack instruction: " + inst);
+                            }
+                        }
+                        case BranchInstruction inst when inst.opcode().isUnconditionalBranch() -> {
+                            BytecodeBasicBlock succ = bcb.successors.get(0);
+                            Block.Reference sb;
+                            // If the block has block parameters for stack operands then
+                            // pop arguments off the stack and use as successor arguments
+                            if (!b.parameters().isEmpty()) {
+                                List<Value> args = new ArrayList<>();
+                                for (int x = 0; x < b.parameters().size(); x++) {
+                                    args.add(stack.pop());
+                                }
+                                sb = c.blockMap.get(succ).successor(args);
+                            } else {
+                                sb = c.blockMap.get(succ).successor();
+                            }
+                            stack.push(b.op(CoreOps.branch(sb)));
```

Comment:
No need to push the result of a terminal operation, since the operation itself also never pushes onto the stack. Likely harmless though, since the stack is localized to each block and such instructions will be the last in the block. See also other cases.